mackerel/host.go

```diff
@@ -10,10 +10,11 @@ type Host struct {
 
 // HostSpec is host specifications sent Mackerel server per hour
 type HostSpec struct {
-	Name          string                 `json:"name"`
-	Meta          map[string]interface{} `json:"meta"`
-	Interfaces    interface{}            `json:"interfaces"`
-	RoleFullnames []string               `json:"roleFullnames"`
-	Checks        []string               `json:"checks"`
-	DisplayName   string                 `json:"displayName,omitempty"`
+	Name             string                 `json:"name"`
+	Meta             map[string]interface{} `json:"meta"`
+	Interfaces       interface{}            `json:"interfaces"`
+	RoleFullnames    []string               `json:"roleFullnames"`
+	Checks           []string               `json:"checks"`
+	DisplayName      string                 `json:"displayName,omitempty"`
+	CustomIdentifier *string                `json:"customIdentifier,omitempty"`
```

Comment:
Thanks, you've got a point. I really dislike the code regarding the empty string as nothing so I wanted to return nil when the custom identifier is not available. But I didn't notice the behaviour of `omitempty` you pointed out, `string` is surely preferable in this case.